### PR TITLE
Fix first-launch-after-install issue (both platforms)

### PR DESCRIPTION
**Pull request by ForsHant**
[Pull request 96](https://github.com/chemerisuk/cordova-plugin-firebase-dynamiclinks/pull/96)

See #96 regarding PR full info.

This pull request is both an Android and iOS implementation of @misterwell solution therefore full fix PR.

### CODE REVIEW

**Comment by ForsHant** [comment]
@chemerisuk merged with master

File: src/android/FirebaseDynamicLinksPlugin.java [hunk 1]
```diff
@@ -91,6 +91,11 @@ public JSONObject then(Task<PendingDynamicLinkData> task) throws JSONException {
                         result.put("deepLink", data.getLink());
                         result.put("clickTimestamp", data.getClickTimestamp());
                         result.put("minimumAppVersion", data.getMinimumAppVersion());
+                        
+			// Hathaway code (Java version) change made to avoid a "phantom" empty deeplink from clobbering a valid deeplink on first launch. Refer to ticket PRG-1613
+			if (data.getLink().equals("")){
```
Comment (chemerisuk) [inline src/android/FirebaseDynamicLinksPlugin.java]: Let's rewrite and simplify this block:
```java
if ("".equals(data.getLink())) return null;
```

**Review by chemerisuk — CHANGES_REQUESTED** [review]

**Comment by chemerisuk** [comment]
@ForsHant take a look at my code suggestion. Also please move this check before line 90

**Comment by ForsHant** [comment]
@chemerisuk Nice suggestion. Added it :)

**Comment by chemerisuk** [comment]
@ForsHant 👍

**Pull request merged**